File: src/main/java/net/softbell/bsh/dto/view/general/ReservSummaryCardDto.java
package net.softbell.bsh.dto.view.general;

import lombok.Getter;
import lombok.Setter;
import net.softbell.bsh.domain.EnableStatusRule;
import net.softbell.bsh.domain.entity.NodeReserv;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 예약뷰 카드정보 DTO
 */
@Getter
@Setter
public class ReservSummaryCardDto
{
	private Long reservId;
	private String description;
	private boolean enableStatus;
	
	public ReservSummaryCardDto(NodeReserv entity)
	{
		// Exception
		if (entity == null)
			return;
		
		// Convert
		this.reservId = entity.getReservId();
		this.description = entity.getDescription();
		
		if (entity.getEnableStatus() == EnableStatusRule.ENABLE)
			this.enableStatus = true;
		else
			this.enableStatus = false;
	}
}

File: src/main/kotlin/net/softbell/bsh/dto/view/general/ReservSummaryCardDto.kt
package net.softbell.bsh.dto.view.general

import net.softbell.bsh.domain.EnableStatusRule
import net.softbell.bsh.domain.entity.NodeReserv

/**
 * @author : Bell(bell@softbell.net)
 * @description : 예약뷰 카드정보 DTO
 */
class ReservSummaryCardDto(entity: NodeReserv) {
    val reservId: Long = entity.reservId
    val description: String = entity.description
    val enableStatus: Boolean = entity.enableStatus == EnableStatusRule.ENABLE
    val creatorNickname: String = entity.member.nickname
    val expression: String = entity.expression
}